File: app/src/main/java/ru/com/gid/SettingsActivity.java
package ru.com.gid;

import android.os.Bundle;
import android.os.PersistableBundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentActivity;

public class SettingsActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        getSupportFragmentManager()
                .beginTransaction()
                .add(R.id.nav_host_fragment, new SettingsFragment())
                .commit();
        setContentView(R.layout.template_main);
    }
}

File: app/src/main/java/ru/com/gid/SettingsActivity.kt
package ru.com.gid

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        supportFragmentManager.beginTransaction().replace(R.id.nav_host_fragment_container,
            SettingsFragment()).commit()
        setContentView(R.layout.activity_settings)
    }
}
